packages/client/ui/react-ui/src/components/embed/v3/crypto/CryptoWalletConnectionHandler.tsx

```diff
@@ -0,0 +1,87 @@
+import DynamicContextProviderWrapper from "@/components/dynamic-xyz/DynamicContextProviderWrapper";
+import type { EmbeddedCheckoutV3IFrameEmitter } from "@crossmint/client-sdk-base";
+import { EthereumWalletConnectors } from "@dynamic-labs/ethereum";
+import { useDynamicContext } from "@dynamic-labs/sdk-react-core";
+import { SolanaWalletConnectors } from "@dynamic-labs/solana";
+import { useEffect } from "react";
+
+export function CryptoWalletConnectionHandler(props: { iframeClient: EmbeddedCheckoutV3IFrameEmitter | null }) {
+    const { iframeClient } = props;
+
+    return (
+        <DynamicContextProviderWrapper
+            settings={{
+                walletConnectors: [EthereumWalletConnectors, SolanaWalletConnectors],
+                events: {
+                    onAuthFlowCancel() {
+                        console.log("[CryptoWalletConnectionHandler] onAuthFlowCancel");
+                        iframeClient?.send("crypto:connect-wallet.failed", {
+                            error: "cancelled",
+                        });
+                    },
+                    onAuthFlowClose() {
+                        console.log("[CryptoWalletConnectionHandler] onAuthFlowClose");
+                    },
+                    onAuthFailure(data, reason) {
+                        console.error("[CryptoWalletConnectionHandler] onAuthFailure", data, reason);
+                    },
+                    onAuthSuccess(data) {
+                        console.log("[CryptoWalletConnectionHandler] onAuthSuccess", data);
+                    },
+                },
+                handlers: {
+                    // biome-ignore lint/suspicious/useAwait: don't neeed to use any await, but we need to return a promise
```

Comment:
I don't get this. This is not returning a promise, but a boolean, right?

Reply:
its a promise bc the funciton is async

Reply:
You mean that `dynamic` types it as async right?

Reply:
yeah, so our function has to also have `async`